tests/call/test_file_reader_decorator.py

```diff
@@ -7,22 +7,30 @@
 
 import os
 import pytest
+import shutil
+import tempfile
 
 import fuzzinator
 
 from common_call import blinesep, resources_dir, MockAlwaysFailCall, MockNeverFailCall
 
 
-@pytest.mark.parametrize('call_init_kwargs, call_kwargs', [
-    ({'init_foo': 'init_bar'}, {'test': os.path.join(resources_dir, 'mock_tests', 'bar.txt')})
+@pytest.mark.parametrize('call_init_kwargs, call_kwargs, decorator_init_kwargs', [
+    ({'init_foo': 'init_bar'}, {'test': os.path.join(resources_dir, 'mock_tests', 'bar.txt')}, {'cleanup': False}),
+    ({'init_foo': 'init_bar'}, {'test': os.path.join(resources_dir, 'mock_tests', 'bar.txt')}, {'cleanup': True})
 ])
 @pytest.mark.parametrize('call_class, exp', [
     (MockAlwaysFailCall, {'init_foo': 'init_bar', 'test': b'bar' + blinesep, 'filename': 'bar.txt'}),
     (MockNeverFailCall, None),
 ])
-def test_file_reader_decorator(call_class, call_init_kwargs, call_kwargs, exp):
-    call_class = fuzzinator.call.FileReaderDecorator()(call_class)
+def test_file_reader_decorator(call_class, call_init_kwargs, decorator_init_kwargs, call_kwargs, exp):
+    call_class = fuzzinator.call.FileReaderDecorator(**decorator_init_kwargs)(call_class)
     call = call_class(**call_init_kwargs)
 
-    with call:
-        assert call(**call_kwargs) == exp
+    with tempfile.TemporaryDirectory() as tempdir:
+        shutil.copy(call_kwargs['test'], tempdir)
+        test_path = os.path.join(tempdir, os.path.basename(call_kwargs['test']))
```

Comment:
`shutil.copy` returns the path to the newly created file, so this can be simplified to

```py
test_path = shutil.copy(call_kwargs['test'], tempdir)
```